rttest/src/rttest.cpp

```diff
@@ -58,17 +60,28 @@ class rttest_sample_buffer
       this->buffer_size = new_buffer_size;
       this->latency_samples = static_cast<int *>(
         std::malloc(new_buffer_size * sizeof(int)));
-      // TODO(dirk-thomas) this must check if the memory was allocated, same below
+      if (!this->latency_samples) {
+        fprintf(stderr, "Failed to allocate latency samples buffer\n");
+        abort();
```

Comment:
If I added exceptions I believe this library could still be linked to C programs, but there is no way of catching the exception from the C program calling this code.